llvm/test/Transforms/InstCombine/sub-gep.ll

```diff
@@ -285,6 +315,98 @@ define i16 @test25_as1(ptr addrspace(1) %P, i64 %A) {
+; CHECK-NEXT:  ret i64 %D
+  %A = getelementptr nusw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i32
+  %C = zext i32 %B to i64
+  %D = sub i64 %C, ptrtoint (ptr addrspace(2) @Arr_as2 to i64)
+  ret i64 %D
+}
+
+define i64 @zext_ptrtoint_sub_ptrtoint_as2_nuw(i32 %offset) {
+; CHECK-LABEL: @zext_ptrtoint_sub_ptrtoint_as2_nuw(
+; CHECK-NEXT:  %A.idx = shl nuw i32 %offset, 1
+; CHECK-NEXT:  %D = zext i32 %A.idx to i64
+; CHECK-NEXT:  ret i64 %D
+  %A = getelementptr nuw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i32
+  %C = zext i32 %B to i64
+  %D = sub i64 %C, ptrtoint (ptr addrspace(2) @Arr_as2 to i64)
+  ret i64 %D
+}
+
+define i64 @zext_ptrtoint_sub_ptrtoint_as2_nusw_nuw(i32 %offset) {
+; CHECK-LABEL: @zext_ptrtoint_sub_ptrtoint_as2_nusw_nuw(
+; CHECK-NEXT:  %A.idx = shl nuw nsw i32 %offset, 1
+; CHECK-NEXT:  %D = zext nneg i32 %A.idx to i64
+; CHECK-NEXT:  ret i64 %D
+  %A = getelementptr nusw nuw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i32
+  %C = zext i32 %B to i64
+  %D = sub i64 %C, ptrtoint (ptr addrspace(2) @Arr_as2 to i64)
+  ret i64 %D
+}
+
+define i64 @zext_ptrtoint_sub_zext_ptrtoint_as2(i32 %offset) {
+; CHECK-LABEL: @zext_ptrtoint_sub_zext_ptrtoint_as2(
+; CHECK-NEXT:  %A.idx = shl nsw i32 %offset, 1
+; CHECK-NEXT:  %E = sext i32 %A.idx to i64
+; CHECK-NEXT:  ret i64 %E
+  %A = getelementptr nusw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i32
+  %C = zext i32 %B to i64
+  %D = zext i32 ptrtoint (ptr addrspace(2) @Arr_as2 to i32) to i64
+  %E = sub i64 %C, %D
+  ret i64 %E
+}
+
+define i64 @zext_ptrtoint_sub_zext_ptrtoint_as2_nuw(i32 %offset) {
+; CHECK-LABEL: @zext_ptrtoint_sub_zext_ptrtoint_as2_nuw(
+; CHECK-NEXT:  %A.idx = shl nuw i32 %offset, 1
+; CHECK-NEXT:  %E = zext i32 %A.idx to i64
+; CHECK-NEXT:  ret i64 %E
+  %A = getelementptr nuw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i32
+  %C = zext i32 %B to i64
+  %D = zext i32 ptrtoint (ptr addrspace(2) @Arr_as2 to i32) to i64
+  %E = sub i64 %C, %D
+  ret i64 %E
+}
+
+define i64 @negative_zext_ptrtoint_sub_ptrtoint_as2_nuw(i32 %offset) {
+  ; CHECK-LABEL: @negative_zext_ptrtoint_sub_ptrtoint_as2_nuw(
+  ; CHECK-NEXT:  %A = getelementptr nuw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  ; CHECK-NEXT:  %1 = ptrtoint ptr addrspace(2) %A to i32
+  ; CHECK-NEXT:  %B.mask = and i32 %1, 65534
+  ; CHECK-NEXT:  %C = zext nneg i32 %B.mask to i64
+  ; CHECK-NEXT:  %D = sub nsw i64 %C, ptrtoint (ptr addrspace(2) @Arr_as2 to i64)
+  ; CHECK-NEXT:  ret i64 %D
+  %A = getelementptr nuw bfloat, ptr addrspace(2) @Arr_as2, i32 %offset
+  %B = ptrtoint ptr addrspace(2) %A to i16
+  %C = zext i16 %B to i64
+  %D = sub i64 %C, ptrtoint (ptr addrspace(2) @Arr_as2 to i64)
+  ret i64 %D
+}
+
+define i64 @ptrtoint_sub_zext_ptrtoint_as2(i32 %offset) {
```

Comment:
```suggestion
define i64 @zext_ptrtoint_sub_ptrtoint_as2_inbounds(i32 %offset) {
```
To follow the rest of your naming convention, I think? I'd also move it up to the other tests like that.

And while there, we can also add a negative test where the GEP does not have any nowrap flags.